src/data/repositories/InstanceDefaultRepository.ts

```diff
@@ -1,25 +1,116 @@
-import { D2Api } from "@eyeseetea/d2-api/2.34";
-import { FutureData } from "../../domain/entities/Future";
-import { User } from "../../domain/entities/User";
+import { D2Api, D2OrganisationUnitSchema, D2UserGroupSchema, SelectedPick } from "@eyeseetea/d2-api/2.34";
+
+import { FutureData, Future } from "../../domain/entities/Future";
+import { GlassModule } from "../../domain/entities/GlassModule";
+import { OrgUnitAccess, UserAccessInfo, ModuleAccess } from "../../domain/entities/User";
 import { InstanceRepository } from "../../domain/repositories/InstanceRepository";
 import { cache } from "../../utils/cache";
 import { getD2APiFromInstance } from "../../utils/d2-api";
 import { apiToFuture } from "../../utils/futures";
+import { DataStoreClient } from "../data-store/DataStoreClient";
+import { DataStoreKeys } from "../data-store/DataStoreKeys";
 import { Instance } from "../entities/Instance";
 
 export class InstanceDefaultRepository implements InstanceRepository {
     private api: D2Api;
 
-    constructor(instance: Instance) {
+    constructor(instance: Instance, private dataStoreClient: DataStoreClient) {
         this.api = getD2APiFromInstance(instance);
     }
 
     public getBaseUrl(): string {
         return this.api.baseUrl;
     }
 
+    mapUserOrgUnitsAccess = (
+        organisationUnits: SelectedPick<
+            D2OrganisationUnitSchema,
+            {
+                id: true;
+                name: true;
+            }
+        >[],
+        dataViewOrganisationUnits: SelectedPick<
+            D2OrganisationUnitSchema,
+            {
+                id: true;
+                name: true;
+            }
+        >[]
+    ): OrgUnitAccess[] => {
+        const orgUnitsAccess = organisationUnits.map(ou => ({
+            orgUnitId: ou.id,
+            orgUnitName: ou.name,
+            readAccess: dataViewOrganisationUnits.findIndex(dvou => dvou.id === ou.id) > -1 ? true : false,
+            captureAccess: true,
+        }));
+
+        //Setting view access for org units that are present in dataViewOrganisationUnits and not organisationUnits
+        dataViewOrganisationUnits.forEach(dvou => {
+            if (orgUnitsAccess.findIndex(ou => ou.orgUnitId === dvou.id) === -1) {
+                orgUnitsAccess.push({
```

Comment:
[functional programming] Here we'd do a filter+map, and concatenate (immutable) with `orgUnitsAccess

Reply:
Addressed in https://github.com/EyeSeeTea/glass-dev/pull/35/commits/2d6776ef32787b23c18581b271cd4d3670d94b21